packages/monaco/src/browser/monaco-snippet-suggest-provider.ts

```diff
@@ -24,17 +24,32 @@ export class MonacoSnippetSuggestProvider implements monaco.modes.ISuggestSuppor
     @inject(FileSystem)
     protected readonly filesystem: FileSystem;
 
-    protected readonly snippets = new Map<string, MonacoSnippetSuggestion[]>();
+    protected readonly snippets = new Map<string, Snippet[]>();
     protected readonly pendingSnippets = new Map<string, Promise<void>[]>();
 
-    async provideCompletionItems(model: monaco.editor.ITextModel): Promise<monaco.modes.ISuggestResult> {
+    async provideCompletionItems(model: monaco.editor.ITextModel, position: monaco.Position, context: monaco.modes.SuggestContext): Promise<monaco.modes.ISuggestResult> {
         const languageId = model.getModeId(); // TODO: look up a language id at the position
         await this.loadSnippets(languageId);
-        const suggestions = this.snippets.get(languageId) || [];
+        const snippetsForLanguage = this.snippets.get(languageId) || [];
+        const suggestions: MonacoSnippetSuggestion[] = [];
+        const word = model.getWordAtPosition(position);
+        const positionEndsInEmptyChar = model.getLineContent(position.lineNumber).substr(0, position.column - 1).match(/\s$/);
+        snippetsForLanguage.forEach(snippet => {
+            if (word && this.isPatternInText(word.word, snippet.prefix, true)) {
```

Comment:
Just checking the current word will lose some of the matches. Snippet's prefix can be multi words.

Reply:
@BroKun thanks for reporting this, Could you show me more information about your multi words test case?

Reply:
![import-single](https://user-images.githubusercontent.com/11978904/63243643-c272ee00-c28d-11e9-8a81-d9f430ec319f.gif)
![import-multi](https://user-images.githubusercontent.com/11978904/63243684-e2a2ad00-c28d-11e9-99b2-56728afaf615.gif)

Reply:
@BroKun thanks, I will take this case nto consideration